### PR TITLE
Ensure variables set by conf.d/fzf.fish are global and update uninstaller

### PR DESCRIPTION
**Pull request by etcusrvar**
I believe you want this to be global.
If `fisher_path` is undefined and this file is in `$__fish_config_dir/conf.d`, it will already be global by default.
If `fisher_path` is defined, and `$fisher_path/conf.d` has to be sourced manually, it will default to local, which I believe is not intended.

### CODE REVIEW

**Comment by PatrickF1** [comment]
Hi @etcusrvar, thanks for contributing a bug fix! I'm afraid I don't understand. What does `fisher_path` have to do with this?

**Comment by etcusrvar** [comment]
I probably should have tested this a bit more before opening this issue.
TL;DR - it's a local setup issue-- but explicitly setting global may still be good idea.

In my local setup, I essentially follow [this example](https://github.com/jorgebucaran/fisher/issues/640#issue-771658021)-- but not exactly.
Instead of
```fish
for file in $fisher_path/conf.d/*.fish
    source $file
end
```
I have
```fish
function __source_fisher_confd
    for file in $fisher_path/conf.d/*.fish
        source $file
    end
end
__source_fisher_confd
```

In the first case, the default scope when sourcing conf.d/fzf.fish is global, but in the second case, being wrapped in a function, the default scope is local.

I have a few other fisher plugins installed that also set variables, but they explicitly set globals where appropriate, so I only saw this issue with your plugin. Even though the issue is with my setup, you may still want to explicitly use --global for `FZF_DEFAULT_OPTS`. (And `__fzf_search_vars_cmd`, which I can add to this PR if desired.)

**Comment by PatrickF1** [comment]
Ohhh I see....I finally understand what the `set --help` docs mean:
> If  a  variable  is  not  explicitly  set  to be either universal, global or local and has never before been
  defined, the variable will be local to the currently executing function. Note that this  is  different  from
  using  the  -l  or --local flag. If one of those flags is used, the variable will be local to the most inner
  currently executing block, while without these the variable will be local to the function. If no function is
  executing, the variable will be global.

Yes, making those explicitly global is a great idea. Thanks @etcusrvar !

File: conf.d/fzf.fish [hunk 1]
```diff
@@ -49,5 +49,7 @@ function _fzf_uninstall --on-event fzf_uninstall
         echo "fzf.fish key bindings removed"
         set_color normal
     end
+
+    set --erase __fzf_search_vars_cmd
```
Comment (PatrickF1) [inline conf.d/fzf.fish:53]: Out of scope change: while updating `__fzf_search_vars_cmd` I realized I never accounted for it in the uninstaller.

**Review by PatrickF1 — APPROVED** [review]

**Comment by etcusrvar** [comment]
Cool. Thanks!
